docs/operate/reference/advanced-modules/_index.md

```diff
@@ -46,6 +46,74 @@ Running {{< glossary_tooltip term_id="modular-resource" text="modular resources"
 
 However, if you are unable to use modular resources because you need to host `viam-server` on a non-Linux system or have an issue with compilation, you may need to [implement a custom component and register it on a server configured as a remote](/operate/reference/advanced-modules/custom-components-remotes/) on your machine.
 
+## Package and deploy using Docker
+
+In rare cases, you may need to package and deploy a module using Docker.
+Use cases for this include:
+
+- Your module has complex system dependencies that cannot be installed on a machine.
+- You have a large bundle and want to use layer caching to reduce the size of the download.
+- You have specific security requirements that are difficult to meet with the default module deployment.
+
+If you choose to deploy your module using Docker, we recommend creating a first run script or binary to run any necessary setup steps.
+
+{{% expand "Click for first run script instructions" %}}
+
+1. Create a tarball that contains:
+
+   - The module's entrypoint script or binary
+   - A <file>meta.json</file>
+   - A first run script or binary that will be executed during the setup phase, for example:
+
+     ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+     #!/usr/bin/env bash
+
+     if [[ -n "$VIAM_TEST_FAIL_RUN_FIRST" ]]; then
+         echo "Sorry, I've failed you."
+         exit 1
+     fi
+
+     docker pull mongo:6
+
+     cat << EOF
+     -------------------------------------
+     Congratulations!
+     The setup script ran successfully!
+     This message is obnoxiously large for
+     testing purposes.
+     Sincerely,
+     First Run Script
+     -------------------------------------
+     EOF
+
+     exit 0
+     ```
+
+   [This example Makefile on GitHub](https://github.com/viam-labs/wifi-sensor/blob/7823b6ad3edcbbbf20b06c34b3181453f5f3f078/Makefile) builds a module binary and bundles it along with a <file>meta.json</file> and first run script.<br><br>
+
+1. Edit the <file>meta.json</file> to include a `first_run` field that points to the first run script or binary.
+
+   ```json
+   {
+     ...
+     "first_run": "first_run.sh"
+   }
+   ```
+
+1. Configure your module on your machine in the same way as you would for a regular module.
+   The first run script will execute once when `viam-server` receives a new configuration.
+   It will only execute once per module or per version of the module.
+
+1. (Optional) After a first run script runs successfully, Viam adds a marker file with a `.first_run_succeeded` suffix in the module’s data directory on disk.
+   It has the location and form: `/root/.viam/packages/data/module/<MODULE_ID>-<VERSION>-<ARCH>.first_run_succeeded`.
```

Comment:
I suspect there should be a slash before `.first_run_succeeded`, but haven't checked on that personally. Maybe it's correct as-is.

Reply:
Hmm also from Maxim's guide--not sure if there's a reasonably practical way to check for sure

Reply:
Eep! I don't have any Docker modules to test this on... @cheukt since it looks like you've been maintaining the confluence instructions, can you confirm?

Reply:
it is `unpackedModDir + FirstRunSuccessSuffix`, an example would be `.viam/packages/data/module/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb-viamrtsp-0_1_0-linux-amd64/bin.first_run_succeeded`
https://github.com/viamrobotics/rdk/blob/1af4d812c36e43c2f08b847290e1642489eb1df8/config/module.go#L279